packages/react/src/prefabs/PreJoin.tsx

```diff
@@ -177,29 +215,42 @@ export const PreJoin = ({
   const [videoEnabled, setVideoEnabled] = React.useState<boolean>(
     defaults.videoEnabled ?? DEFAULT_USER_CHOICES.videoEnabled,
   );
-  const [videoDeviceId, setVideoDeviceId] = React.useState<string>(
-    defaults.videoDeviceId ?? DEFAULT_USER_CHOICES.videoDeviceId,
-  );
+  const initialVideoDeviceId = defaults.videoDeviceId ?? DEFAULT_USER_CHOICES.videoDeviceId;
+  const [videoDeviceId, setVideoDeviceId] = React.useState<string>(initialVideoDeviceId);
+  const initialAudioDeviceId = defaults.audioDeviceId ?? DEFAULT_USER_CHOICES.audioDeviceId;
   const [audioEnabled, setAudioEnabled] = React.useState<boolean>(
     defaults.audioEnabled ?? DEFAULT_USER_CHOICES.audioEnabled,
   );
-  const [audioDeviceId, setAudioDeviceId] = React.useState<string>(
-    defaults.audioDeviceId ?? DEFAULT_USER_CHOICES.audioDeviceId,
-  );
+  const [audioDeviceId, setAudioDeviceId] = React.useState<string>(initialAudioDeviceId);
 
-  const video = usePreviewDevice(videoEnabled, videoDeviceId, 'videoinput');
+  const tracks = usePreviewTracks(
+    {
+      audio: audioEnabled ? { deviceId: initialAudioDeviceId } : false,
+      video: videoEnabled ? { deviceId: initialVideoDeviceId } : false,
+    },
+    onError,
+  );
 
```

Comment:
I dont seem to find where the tracks are updated if `audioDeviceId`/`videoDeviceId` change?
Previously (with `usePreviewDevice` this happened when changing the params of the hook
```ts
const video = usePreviewDevice(videoEnabled, /*HERE:*/ videoDeviceId, 'videoinput');
```

Reply:
Oh. found it. the tracks are now passed to the setupDeviceSelector... That is very reasonable. (Because it mimics the api which is present if a room is available)

Reply:
yeah exactly. the idea is that `usePreviewTracks` is only used for the initial value in order to facilitate a unified permission dialog if both video and microphone are enabled. Subsequent permission requests are then handled on the track level directly. 
I hope this accounts for all edge cases, but it will definitely need some more testing/iterations to be sure